### PR TITLE
feat: Sort conversations list search results - WPB-14955

### PR DESCRIPTION
**Pull request by samwyndham**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-14955" title="WPB-14955" target="_blank">TaskWPB-14955</a>  Improve Search in all Conversations
  </td></table>
  <br />
 

### Issue

This PR adds sorting to conversation list search results so that:

- Matched one-on-one conversations appear at the beginning.
- Conversations where the query matches then name appear in the middle.
- Conversations where the query matches a participant name appear at the end.

### Testing

1. Navigate to the conversations tab.
2. Begin searching
3. Verify, that one-on-one conversations appear first, conversations where the query matches the conversation name appear next, and conversations where the query matches a participant name appear last.  

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneMigrator.swift
Comment (samwyndham) [inline wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneMigrator.swift:1]: This file has only lint and formatting changes

File: wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift
Comment (samwyndham) [inline wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift:1]: This file has only lint and formatting changes

File: wire-ios-sync-engine/Tests/Source/UserSession/ZMUserSessionTestsBase.swift
Comment (samwyndham) [inline wire-ios-sync-engine/Tests/Source/UserSession/ZMUserSessionTestsBase.swift:1]: This file has only lint and formatting changes

**Review by caldrian — APPROVED** [review]

File: wire-ios-data-model/Source/ConversationSearch/ZMFilterableConversationAdapter.swift [hunk 1]
```diff
@@ -37,6 +37,10 @@ public struct ZMFilterableConversationAdapter: FilterableConversation {
             }
     }
 
+    public var isOneOnOne: Bool {
+        conversation.oneOnOneUser != nil
```
Comment (KaterinaWire) [inline wire-ios-data-model/Source/ConversationSearch/ZMFilterableConversationAdapter.swift]: why not `conversation.conversationType == .oneOnOne` ?

**Comment by KaterinaWire** [comment]
@samwyndham I left one small question/suggestion. Otherwise it looks good 👍

File: wire-ios-data-model/Source/ConversationSearch/ZMFilterableConversationAdapter.swift (hunk 1 above)
Comment (samwyndham) [reply wire-ios-data-model/Source/ConversationSearch/ZMFilterableConversationAdapter.swift]: I discussed this a little with @johnxnguyen in the developer channel. My understanding is:

- We should only have at most a single one on one conversation displayed per non self user.
- However it is possible there are multiple conversations with type `. oneOnOne` (duplicates or after migrating to MLS for example).

The `conversation.oneOnOneUser` helps ensure that only one of these conversations is treated as a true one on one and moved to the top of the search results.

To be safe I will add your suggestion and keep the current check:

```swift
conversation.conversationType == .oneOnOne && conversation.oneOnOneUser != nil
```

Comment (samwyndham) [reply wire-ios-data-model/Source/ConversationSearch/ZMFilterableConversationAdapter.swift]: Implemented in 95d2ad8d69956bfc4f1d56d38489ec27eea29cb9

File: wire-ios/Wire-iOS Tests/AudioMessageViewTests.swift
Comment (samwyndham) [inline wire-ios/Wire-iOS Tests/AudioMessageViewTests.swift:1]: Changes in this file are to fix tests warnings when running -  we shouldn't use ManagedObjects without a managed object context.

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**